daemon/execdriver/native/driver.go

```diff
@@ -335,7 +335,9 @@ func (d *driver) Clean(id string) error {
 }
 
 func (d *driver) Stats(id string) (*execdriver.ResourceStats, error) {
+	d.Lock()
 	c := d.activeContainers[id]
+	d.Unlock()
```

Comment:
yeah we talked about this yesterday on IRC but I'd need another PR if this is ok for you, is it?